group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinator.java

```diff
@@ -63,6 +63,11 @@
  */
 public interface GroupCoordinator {
 
+    /**
+     * @return True if the new coordinator; False otherwise.
+     */
+    boolean isNewGroupCoordinator();
```

Comment:
I introduced this new method to the GroupCoordinator interface and I use it at runtime (e.g. in KafkaApi) to determine whether the new or the old coordinator is used. I did not in order to not rely on the `group.coordinator.new.enable` configuration at runtime. The issue was that in ZK mode, the `group.coordinator.new.enable` is also true by default now but the configuration is just ignored when the zk broker starts.